File: projects/m2/012-shuffling-a-deck-of-cards/solutions/jimmyhu/solution-shuffling-a-deck-of-cards.py
import random

def createDeck():
    deck = []
    suits = ['s','h','d','c']
    high_card = ['T','J','Q','K','A']
    for suit in suits:
        for i in range(2,10):
            deck.append(f"{i}{suit}")
        for s in high_card:
            deck.append(f"{s}{suit}")
    return deck

def shuffle(lst):
    randomized = []
    for card in lst:
        position = random.randint(0,51)
        randomized.insert(position,card)
    return randomized
            



def main():
    print(createDeck())
    print(shuffle(createDeck()))
if __name__ == '__main__':
    main()

File: projects/m2/012-shuffling-a-deck-of-cards/valentinaBazzacchi/solution-shuffling-a-deck-of-cards.js
// MAIN PROGRAM
let deck = []; 
console.log(createDeck());

let shuffleDeck;
console.log(shuffle(deck));

// FUNCTIONS
function createDeck() {
    let suits = ["s", "h", "d", "c"];
    let values = ["T", "J", "Q", "K", "A"];
    
    suits.forEach(function(suit){
        for (let i = 2, y = 0; i < 15; i++) {
            
            if (i <= 9) {
                deck.push(i + suit);
            } else {
                deck.push(values[y] + suit);
                y++;
            } // end condition
        
        } // end For loop
    }) // end forEach   
    return deck;
} // end function 

function shuffle(list) {
    shuffleDeck = list;
    
    for (let i = 0; i < shuffleDeck.length; i++) {
        
        let randomIndex = Math.floor(Math.random() * (52 - i)) + i;    
        let currentCard = shuffleDeck[i];
        shuffleDeck[i] = shuffleDeck[randomIndex];
        shuffleDeck[randomIndex] = currentCard;
        
    } // end For loop
    return shuffleDeck;
}// end function